packages/demos/stories/TableColumnsDragDrop.tsx

```diff
@@ -0,0 +1,286 @@
+/**
+ * SPDX-FileCopyrightText: © 2020 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import ClayTable from '@clayui/table';
+import classNames from 'classnames';
+import React from 'react';
+import {
+	DropTargetMonitor,
+	XYCoord,
+	useDrag,
+	useDragLayer,
+	useDrop,
+} from 'react-dnd';
+
+const TABLE_COLUMN_DND_TYPE = 'tableColumn';
+
+const initialTableColumns = [
+	{
+		data: ['Red, white & blue', 'Tricolore', 'Stars & stripes'],
+		id: 'flag',
+		title: 'Flag',
+	},
+	{
+		data: ['Europe', 'Europe', 'North America'],
+		id: 'region',
+		title: 'Region',
+	},
+	{
+		data: ['Croatia', 'Italy', 'USA'],
+		id: 'country',
+		title: 'Country',
+	},
+	{
+		data: [
+			'Lorem ipsum dolor, sit amet consectetur adipisicing elit.',
+			'Lorem ipsum dolor, sit amet consectetur adipisicing elit.',
+			'Lorem ipsum dolor, sit amet consectetur adipisicing elit.',
+		],
+		id: 'description',
+		title: 'Description',
+	},
+];
+
+const dragLayerStyles: React.CSSProperties = {
+	height: '100%',
+	left: 0,
+	pointerEvents: 'none',
+	position: 'fixed',
+	top: 0,
+	width: '100%',
+	zIndex: 100,
+};
+
+export type TColumn = {
+	data: Array<string>;
+	id: string;
+	title: string;
+};
+
+export interface ICustomDragLayerProps {
+	columns: Array<TColumn>;
+}
+
+const CustomDragLayer: React.FunctionComponent<ICustomDragLayerProps> = ({
+	columns,
+}) => {
+	const {currentOffset, initialOffset, isDragging, item} = useDragLayer(
+		monitor => ({
+			currentOffset: monitor.getSourceClientOffset(),
+			initialOffset: monitor.getInitialSourceClientOffset(),
+			isDragging: monitor.isDragging(),
+			item: monitor.getItem(),
+		})
+	);
+
+	if (!isDragging) {
+		return null;
+	}
+
+	const columnBeingDragged = columns.find(
+		(column: TColumn) => column.id === item.id
+	);
+
+	return (
+		<ColumnDragPreview
+			column={columnBeingDragged!}
+			currentOffset={currentOffset}
+			initialOffset={initialOffset}
+		/>
+	);
+};
+
+export interface IColumnDragPreviewProps {
+	column: TColumn;
+	currentOffset: XYCoord | null;
+	initialOffset: XYCoord | null;
+}
+
+const ColumnDragPreview: React.FunctionComponent<
+	IColumnDragPreviewProps
+> = React.memo(({column, currentOffset, initialOffset}) => {
```

Comment:
Whats the thought behind adding `memo`? Does it make a notable difference if we don't use them? Primarily I am thinking of how we can simplify for the sake of others who will look at this demo for an example.

Reply:
Removing the memo worked! It was taken from their examples.